rsmtool/utils.py

```diff
@@ -470,19 +470,27 @@ def partial_correlations(df):
     if numcols > numrows:
         icvx = empty_array
     else:
-        # we also return nans if there is singularity in the data
-        # (e.g. all human scores are the same)
         try:
-            icvx = np.linalg.inv(df_cov)
-        except np.linalg.LinAlgError:
+            # if the determinant is less than the lowest representable
+            # 32 bit integer, then we use the try the pseudo-inverse;
+            # otherwise, we try the inverse; if neither works, we
+            # set the entire matrix to NaNs
+            if np.linalg.det(df_cov) < np.finfo(np.float32).eps:
+                icvx = np.linalg.pinv(df_cov)
+                warnings.warn('The inverse of the variance-covariance matrix '
+                              'was calculated using the Moore-Penrose generalized '
+                              'matrix inversion, due to its determinant being at '
+                              'or very close to zero.')
+            else:
+                icvx = np.linalg.inv(df_cov)
+        except:
```

Comment:
I am not a fan of this particular way of doing this especially since using `except` without a specific exception isn't a good idea. Can't we structure this as a `try ... except` with an `AssertionError` for the determinant and then whatever specific exception you were trying to catch after that?

Reply:
Sure, I can restructure the code this way. I'll also add some more tests.